vm/src/vm/runners/cairo_pie.rs

```diff
@@ -41,6 +41,8 @@ pub type Pages = HashMap<usize, PublicMemoryPage>;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
 pub struct OutputBuiltinAdditionalData {
+    #[serde(skip)]
+    pub base: usize,
```

Comment:
Why is this field needed?

Reply:
The problem is the following: the bootloader starts by saving the VM output builtin and replaces it by a dedicated one to run the "simple" bootloader program and capture its output.

To resolve that problem, I thought about adding this field here so that we can get the full state of the output builtin in one go. We don't want to serialize it, hence the `skip`. But I'm open to other proposals, I could create a dedicated struct for example.

Reply:
Why does the new output builtin need to know the base of the old one?

Reply:
We just store it so that we can restore it later on.

Reply:
The output builtin's segment index can already be found on the `CairoPie` itself on the  CairoPie.metdata.builtin_segments map. I don't see the need to add it in the `BuiltinAdditionalData` too

Reply:
When implementing the bootloader hint there is no `CairoPie` object to store this info. The flow goes like this:

1. VM starts, output builtin is configured
2. Bootloader starts, one of the first steps is to run the "simple" bootloader program and capture its output
3. To achieve this, the first bootloader hint stores the original output builtin configured at step 1 and replaces it with a new one to capture the output of the simple bootloader.
4. After having executed the simple bootloader, the bootloader restores the original output builtin to write its own output.

The simplest way to store the output builtin state was to store this information in the additional data (as is done in the Python VM). I can as easily create a new struct if you prefer it that way.

Reply:
I think there is a mixup in what the output builtin's state is (base, attributes & pages) and what the additional data is (attributes & pages). The python vm has its own `get_state` method that has the behaviour you are adding to `get_additional_data`

Reply:
I can definitely do that, but then the methods to retrieve/restore the output builtin state become more specific. I can either:

1. Replace the whole builtin object
2. Add a method that takes a base + additional data to set the builtin state
3. Add a new struct (ex: `OutputBuiltinState`)  and add a getter + setter for this particular struct.

What do you prefer?

Reply:
2 & 3 are both good solutions, 2 sound simpler so I would go for that one

Reply:
I went with solution 3 and introduced an `OutputBuiltinState` struct.